teps/0081-add-chains-subcommand-to-the-cli.md

```diff
@@ -0,0 +1,326 @@
+
+When working with `Chains`, the user experienced will be enhanced if users are
+able to
+*  use the Tekton CLI rather than a combination of other tools like `kubectl`,
+`jq`, `base64`, etc.
+* not having to memorize which exact TaskRun annotation or key to query.
+* discover available `Chains` features via the CLI and/or its documentation.
+
+### Goals
+
+Add the ability to run `tkn chains <chains_action>` fom the CLI.
+
+### Non-Goals
+
+While the idea is to use the Tekton CLI to replace a combination of shell
+commands, we may not want to implement all exotic variations and focus only on
+the most common use cases in order to avoid overloading the `chains` command.
+
+### Use Cases
+
+#### Use Case 1
+
+Extract the Chains payload from a taskrun:
+
+```bash
+$ tkn chains payload cache-image-pipelinerun-r-qd6xw-build-trusted-jrj5v
+{
+  "conditions": [
+    {
+      "type": "Succeeded",
+      "status": "True",
+      "lastTransitionTime": "2021-08-31T15:19:45Z",
+      "reason": "Succeeded",
+      "message": "All Steps have completed executing"
+    }
+  ],
+  "podName": "cache-image-pipelinerun-r-qd6xw-build-trusted-jrj5v-pod-qptxn",
+  "startTime": "2021-08-31T15:19:17Z",
+  "completionTime": "2021-08-31T15:19:45Z",
+  ...
+}
+```
+
+Instead of:
+
+```bash
+TASKRUN=cache-image-pipelinerun-r-qd6xw-build-trusted-jrj5v
+TASKRUN_UID=$(kubectl get taskrun $TASKRUN -o=json | jq -r '.metadata.uid')
+kubectl get taskrun $TASKRUN -o=json \
+  | jq -r ".metadata.annotations[\"chains.tekton.dev/payload-taskrun-$TASKRUN_UID\"]" \
+  |base64 --decode \
+  | jq
+```
+
+#### Use Case 2
+
+Extract the signature from a taskrun:
+
+```bash
+$ tkn chains signature cache-image-pipelinerun-r-qd6xw-build-trusted-jrj5v
+MEUCIQCHS4in92kQW6B+LfnEilpRcOg5akspPh4mQ+tSaEq/jgIgdl83eCyw41xFC8xti6j0/TgzXkKVixfD30yenabWyHU=
+```
+
+Instead of:
+
+```bash
+TASKRUN=cache-image-pipelinerun-r-qd6xw-build-trusted-jrj5v
+TASKRUN_UID=$(kubectl get taskrun $TASKRUN -o=json | jq -r '.metadata.uid')
+kubectl get taskrun $TASKRUN -o=json \
+  | jq -r ".metadata.annotations[\"chains.tekton.dev/signature-taskrun-$TASKRUN_UID\"]"
+```
+
+#### Use Case 3
+
+Change the provenance format:
+
+```bash
+$ tkn chains format in-toto
+```
+
+instead of:
+
+```bash
+ kubectl patch \
+  configmap chains-config \
+  -n tekton-chains \
+  -p='{"data":{"artifacts.taskrun.format": "in-toto"}}'
+```
+
+#### Other Use Case ideas
+
+* Generate and configure `cosign` keys
+* Verify `cosign` signatures
+* Validate `in-toto` layout
+
+## Requirements
+
+There may not be any specific requirements, depending on the commands being
+implemented. See the [Notes/Caveats (optional)](notes-caveats-optional) for more
+details
+
+## Proposal
+
+```shell
+❯ ./tkn chains --help
+Manage Tekton Chains
+
+Usage:
+tkn chains [flags]
+tkn chains [command]
+
+Available Commands:
+  payload     Print a Tekton chains' payload for a specific taskrun
+  signature   Print a Tekton chains' signature for a specific taskrun
```

Comment:
We could add a `tkn chains config` command for modifying the configmap as you mentioned above! 

```
tkn chains config set <key> <value>
tkn chains config unset <key> <value>
```

could be options

Reply:
Good option! 

I am a bit hesitant (but not against idea!) since we are trying to keep the CLI simple and high level. Therefore this would potential fit in the "non-goals" category.

However, if we want to provide a simple way to modify any of the values described in the [documentation](https://github.com/tektoncd/chains/blob/main/docs/config.md), then this is definitely the way to go!

Reply:
:+1: